redgenes/bakta_annotations.py

```diff
@@ -0,0 +1,189 @@
+    """
+    Annotate genomes based on Bakta and insert information into the database.
+    """
+    for _, row in df.iterrows():
+        local_path, filename = Path((row["local_path"])), row["assembly_accession"]
+        source, external_accession = row["source"], row["assembly_accession"]
+        with copy_and_unzip(str(local_path), tmpdir) as input_fasta:
+            bakta_output_tsv, bakta_commands = run_bakta(input_fasta, tmpdir, bakta_db)
+        bakta_df = extract_bakta_results(str(bakta_output_tsv))
+
+        with TRN:
+            sql_identifier = """
+                INSERT INTO identifier (filename_full, filepath)
+                VALUES (?, ?)
+                RETURNING entity_id"""
+            args_identifer = [filename, str(local_path)]
+            TRN.add(sql_identifier, args_identifer)
+            entity_id = TRN.execute_fetchflatten()
+
+            sql_md_info = """
+                INSERT INTO md_info (entity_id, source, external_accession)
+                VALUES (?, ?, ?)"""
+            args_md_info = [source, external_accession]
+            args_md_info = entity_id + args_md_info
+            TRN.add(sql_md_info, args_md_info)
+
+            sql_extract_bakta_run_info = """
+                SELECT run_id FROM run_info
+                WHERE software = 'bakta' and version = ? and commands = ?"""
+            TRN.add(
+                sql_extract_bakta_run_info, [bakta_version, bakta_commands]
+            )
+            bakta_run_id = TRN.execute_fetchflatten()
+
+            if not bakta_run_id:
+                sql_run_info_bakta = """
+                    INSERT INTO run_info (software, version, commands)
+                    VALUES (?, ?, ?) RETURNING run_id"""
+                args_run_info_bakta = [
+                    "bakta",
+                    bakta_version,
+                    bakta_commands,
+                ]
+                TRN.add(sql_run_info_bakta, args_run_info_bakta)
+                bakta_run_id = TRN.execute_fetchflatten()
+
+            bakta_df.insert(0, "entity_id", entity_id[0])
+            bakta_df.insert(9, "run_accession", bakta_run_id[0])
+            args_bakta_info = bakta_df.iloc[:, 0:10].values.tolist()
+            sql_bakta_info = """
+                INSERT INTO bakta (
+                    entity_id,
+                    contig_id,
+                    type,
+                    start,
+                    stop,
+                    strand,
+                    locus_tag,
+                    gene,
+                    product,
+                    run_accession)
+                VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+                RETURNING bakta_accession;"""
+            TRN.add(sql_bakta_info, args_bakta_info, many=True)
+            bakta_accession = TRN.execute()
+            nested_list = [item for sublist in bakta_accession if sublist for item in sublist]
+            bakta_df['bakta_accession'] = [item[0] for item in nested_list[2:] if item]
```

Comment:
I'm not sure I understand this

Reply:
To be honest, I did this because bakta_accession was not returning what I wanted it to and this is the only way I could find to get it to work. I'm sure there is a better way, but I don't know what it would be. I'm trying to return the bakta accession for each gene that I am entering into the database.